pkg/controller/core/cohort_controller_test.go

```diff
@@ -0,0 +1,294 @@
+			{Flavor: "red", Resource: "cpu"}: 10_000,
+		}
+		if diff := cmp.Diff(wantQuotas, cohortSnap.ResourceNode.SubtreeQuota); diff != "" {
+			t.Fatalf("unexpected quota (-want +got) %s", diff)
+		}
+	}
+
+	// update
+	{
+		if err := cl.Get(ctx, client.ObjectKeyFromObject(cohort), cohort); err != nil {
+			t.Fatal("unexpected error")
+		}
+		cohort.Spec.ResourceGroups[0] = utiltesting.ResourceGroup(
+			utiltesting.MakeFlavorQuotas("red").Resource("cpu", "5").FlavorQuotas,
+		)
+		if err := cl.Update(ctx, cohort); err != nil {
+			t.Fatal("unexpected error updating cohort", err)
+		}
+		if _, err := reconciler.Reconcile(
+			ctx,
+			reconcile.Request{NamespacedName: client.ObjectKeyFromObject(cohort)},
+		); err != nil {
+			t.Fatal("unexpected error")
+		}
+
+		cohortSnap, ok := cache.Snapshot().Cohorts["cohort"]
+		if !ok {
+			t.Fatal("expected Cohort in snapshot")
+		}
+
+		wantQuotas := resources.FlavorResourceQuantities{
+			{Flavor: "red", Resource: "cpu"}: 5_000,
+		}
+		if diff := cmp.Diff(wantQuotas, cohortSnap.ResourceNode.SubtreeQuota); diff != "" {
+			t.Fatalf("unexpected quota (-want +got) %s", diff)
+		}
+	}
+
+	// delete
+	{
+		if err := cl.Delete(ctx, cohort); err != nil {
+			t.Fatal("unexpected error during deletion")
+		}
+		if _, err := reconciler.Reconcile(
+			ctx,
+			reconcile.Request{NamespacedName: client.ObjectKeyFromObject(cohort)},
+		); err != nil {
+			t.Fatal("unexpected error")
+		}
+
+		if _, ok := cache.Snapshot().Cohorts["cohort"]; ok {
+			t.Fatal("unexpected Cohort in snapshot")
+		}
+	}
+}
+
+func TestCohortReconcilerFilters(t *testing.T) {
+	cl := utiltesting.NewClientBuilder().
+		Build()
+	cache := cache.New(cl)
+	qManager := queue.NewManager(cl, cache)
+	reconciler := NewCohortReconciler(cl, cache, qManager)
+
+	t.Run("delete returns true", func(t *testing.T) {
+		if !reconciler.Delete(event.DeleteEvent{}) {
+			t.Fatal("expected delete to return true")
+		}
+	})
+
+	t.Run("create returns true", func(t *testing.T) {
+		if !reconciler.Create(event.CreateEvent{}) {
+			t.Fatal("expected create to return true")
+		}
+	})
+
+	t.Run("generic returns true", func(t *testing.T) {
+		if !reconciler.Generic(event.GenericEvent{}) {
+			t.Fatal("expected generic to return true")
+		}
+	})
+
+	t.Run("update provided wrong type", func(t *testing.T) {
+		cohort := utiltesting.MakeCohort("cohort").Cohort
+		nonCohort := utiltesting.MakeClusterQueue("cq").ClusterQueue
+		t.Run("old wrong type returns false", func(t *testing.T) {
+			event := event.UpdateEvent{
+				ObjectOld: &nonCohort,
+				ObjectNew: &cohort,
+			}
+			if reconciler.Update(event) {
+				t.Fatal("expected false")
+			}
+		})
+		t.Run("new wrong type returns false", func(t *testing.T) {
+			event := event.UpdateEvent{
+				ObjectOld: &cohort,
+				ObjectNew: &nonCohort,
+			}
+			if reconciler.Update(event) {
+				t.Fatal("expected false")
+			}
+		})
+	})
+	t.Run("update", func(t *testing.T) {
+		cohort := utiltesting.MakeCohort("cohort").Cohort
```

Comment:
This could be converted into a test table, like so:

```golang
tests := []{
  oldObj runtime.Object
  newObj runtime.Object
  want bool
} {...}
```

And even the "update provided wrong type" can be part of the table.